api/src/test/java/org/pmiops/workbench/monitoring/GaugeRecorderServiceTest.java

```diff
@@ -0,0 +1,102 @@
+package org.pmiops.workbench.monitoring;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.verify;
+
+import com.google.common.collect.ImmutableMap;
+import java.time.Clock;
+import java.util.Collections;
+import java.util.Map;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.pmiops.workbench.billing.BillingProjectBufferService;
+import org.pmiops.workbench.cohortreview.CohortReviewService;
+import org.pmiops.workbench.monitoring.views.MonitoringViews;
+import org.pmiops.workbench.monitoring.views.OpenCensusStatsViewInfo;
+import org.pmiops.workbench.test.FakeClock;
+import org.pmiops.workbench.workspaces.WorkspaceServiceImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+public class GaugeRecorderServiceTest {
+
+  private static final Map<OpenCensusStatsViewInfo, Number> BILLING_BUFFER_GAUGE_MAP =
+      ImmutableMap.of(
+          MonitoringViews.BILLING_BUFFER_ASSIGNING_PROJECT_COUNT,
+          2L,
+          MonitoringViews.BILLING_BUFFER_AVAILABLE_PROJECT_COUNT,
+          1L,
+          MonitoringViews.BILLING_BUFFER_CREATING_PROJECT_COUNT,
+          0L);
+  private static final long WORKSPACES_COUNT = 101L;
+  private static final Map<OpenCensusStatsViewInfo, Number> WORKSPACE_GAUGE_MAP =
```

Comment:
TOL: If we're testing the WORKSPACE_GAUGE_MAP and the BILLING_BUFFER_GAUGE_MAP, does it make sense to test the other maps? Or is the goal here to just test a couple of gauges?

I think it's fine to just test a couple of gauges, so we know it works and don't add too much overhead to adding more?

Reply:
The point of multiple maps is to test the map streaming/joining logic and demonstrate how the injection scheme works. It's not mean to be all-inclusive.

For a simple enum, you could declare a local test enum, like `TestGauges.GAUGE_A`, but I think that would be hairy in this case (since all the entries have to satisfy the parent interface).